packages/compiler-core/src/transforms/transformElement.ts

```diff
@@ -740,10 +740,17 @@ export function buildProps(
             // so use `hasStyleBinding` to ensure that it is a dynamic style binding
             hasStyleBinding
           ) {
-            styleProp.value = createCallExpression(
-              context.helper(NORMALIZE_STYLE),
-              [styleProp.value]
-            )
+            const content = (styleProp.value as any).content
+            // #4138 only transform object style
+            if (
+              content &&
+              (content.startsWith('{') || content.startsWith('['))
```

Comment:
This won't cover cases where the object, array or string come from a instance property or other expression, I think:
```html
<div :style"thisReturnsAnArray">
```

Wouldn't it be better to handle this in the `normalizeStyle` helper?